iterativeTrimParallel.py

```diff
@@ -0,0 +1,593 @@
+                vfatTree.ReadFile(args.vfatConfig)
+                
+                for event in vfatTree:
+                    # Skip masked vfats
+                    if (args.vfatmask >> int(event.vfatN)) & 0x1:
+                        continue
+                    
+                    # Write CFG_THR_ARM_DAC
+                    print('Set link {0} VFAT{1} CFG_THR_ARM_DAC to {2}'.format(args.link,event.vfatN,event.vt1))
+                    vfatBoard.setVFATThreshold(chip=int(event.vfatN), vt1=int(event.vt1))
+                    
+        # Get all chip IDs
+        vfatIDvals = vfatBoard.getAllChipIDs(args.vfatmask)
+
+        # Get CFG_CAL_DAC calibration constants
+        if args.calFileCAL:
+            calDacCalFile = "{0}/{1}/calFile_calDac_{1}.txt".format(dataPath,chamber_config[ohKey])
+            calDacCalFileExists = os.path.isfile(calDacCalFile)
+            if not calDacCalFileExists:
+                printRed("Missing CFG_CAL_DAC calibration file for shelf{0} slot{1} OH{2}, detector {3}:\n\t{4}".format(
+                    args.shelf,
+                    args.slot,
+                    ohN,
+                    chamber_config[ohKey],
+                    calDacCalFile))
+                raise Exception("Missing CFG_CAL_DAC Calibration file.")
+
+            ohKey = (args.shelf,args.slot,ohN)            
+            dict_calFiles[ohKey] = calDacCalFile
+            pass
+        
+    # Get initial channel registers
+    if args.zeroChan:
+        print("zero'ing all channel registers on shelf{0} slot{1} in ohMask: 0x{2:x}".format(args.shelf,args.slot,args.ohMask))
+        for ohN in range(0,amcBoard.nOHs+1):
+            # Skip masked OH's
+            if( not ((args.ohMask >> ohN) & 0x1)):
+                continue
+            vfatBoard = HwVFAT(cardName, ohN, args.debug)
+            rpcResp = vfatBoard.setAllChannelRegisters(vfatMask=args.vfatmask)
+                    
+            if rpcResp != 0:
+                raise Exception("{0}RPC response was non-zero, zero'ing all channel registers failed{1}".format(colors.RED,colors.ENDC))
+        pass
+
+    dataType=[
+            ('CALPULSE_ENABLE','uint8'),
+            ('MASK','uint8'),
+            ('ZCC_TRIM_POLARITY','uint8'),
+            ('ZCC_TRIM_AMPLITUDE','uint8'),
+            ('ARM_TRIM_POLARITY','uint8'),
+            ('ARM_TRIM_AMPLITUDE','uint8')]
+
+    dict_chanRegArray = {}
+    dict_dfTrimResults = {}
+
+    # Set a dummy mask reason since we are not using the full scurve analysis
+    maskTransform = lambda x: (x & 0x1) << 5
+
+    # Make containers for tracking iteration progress
+    scurveStatTypes=[ ("iterN","uint8"),("vfatN","uint8"),("vfatID","uint32"),("avg","f4"),("std","f4"),("max","f4"),("min","f4"),("p2p","f4"),("n_trimmed","uint8") ]
+
+    dict_chanRegArray = {}
+    dict_chanRegArray[0] = {}
+    
+    #initialize bookkeeping information
+    for ohN in range(0,amcBoard.nOHs+1):
```

Comment:
Why do we have `amcBoard.nOHs+1` as a top end of range? This adds an extra iteration (which doesn't mean anything as such OH will always be masked)

Reply:
I don't see any reason for this `+1` either. I fixed this here and in all other places in `vfatqc-python-scripts`.